cli/src/main.rs

```diff
@@ -49,20 +49,40 @@ async fn main() -> Result<()> {
         (@arg cache: +takes_value +required "cache location"))
       (@subcommand certs =>
         (@arg domain: +takes_value +required "FQDN of the cluster"))
-    );
+    )
+    .subcommand(<Opts as IntoApp>::into_app().name("deploy"));
 
     let mut help_text = Vec::new();
     app.write_help(&mut help_text)
         .expect("Failed to write help text to buffer");
     let matches = app.get_matches();
 
     match matches.subcommand() {
-        Some(("rebuild", sub)) => cli::rebuild(sub).await,
-        Some(("info", sub)) => cli::info(sub).await,
-        Some(("ssh", sub)) => cli::ssh(sub).await,
-        Some(("terraform", sub)) => cli::terraform(sub).await,
-        Some(("provision", sub)) => cli::provision(sub).await,
-        Some(("certs", sub)) => cli::certs(sub).await,
+        Some(("rebuild", sub)) => {
+            pretty_env_logger::init();
+            cli::rebuild(sub).await
+        }
+        Some(("deploy", sub)) => cli::deploy(sub).await,
+        Some(("info", sub)) => {
+            pretty_env_logger::init();
+            cli::info(sub).await
+        }
+        Some(("ssh", sub)) => {
+            pretty_env_logger::init();
+            cli::ssh(sub).await
+        }
+        Some(("terraform", sub)) => {
+            pretty_env_logger::init();
+            cli::terraform(sub).await
+        }
+        Some(("provision", sub)) => {
+            pretty_env_logger::init();
+            cli::provision(sub).await
+        }
+        Some(("certs", sub)) => {
+            pretty_env_logger::init();
+            cli::certs(sub).await
+        }
```

Comment:
This hurts my eyes explain this 🤣

Reply:
I'm assuming that we're getting double logging from `deploy-rs`?

Reply:
Ok yeah I'd say you should move the logger initialization out of the "lib part" https://github.com/nrdxp/deploy-rs/blob/as-lib/src/cli.rs#L616-L620
This has to be at the very top of the actual `main()` function cause this is an app concern not a lib concern.

Reply:
yeah it's annoying, but deploy-rs init's it's own logger and it panics if I don't do this. I'm not sure if our logger and theirs is compatible so I did it this way for now

Reply:
So after trying it, it doesn't work unfortunately. It causes deploy-rs to not display important output during a deployment. We might be able to make this a bit nicer with a function abstraction or even a macro of some kind, but other than that, it'll have to stay this way semantically.